addons/mail/models/mail_message.py

```diff
@@ -140,6 +140,7 @@ def default_get(self, fields):
     # Besides for new messages, and messages never sending emails, there was no mail, and it was searching for nothing.
     mail_ids = fields.One2many('mail.mail', 'mail_message_id', string='Mails', groups="base.group_system")
     canned_response_ids = fields.One2many('mail.shortcode', 'message_ids', string="Canned Responses", store=False)
+    open_thread = fields.Boolean(default=True, readonly=True)
```

Comment:
mail.message is an heavily used table, both for content (html fields, lot of stored information) and for row count. Adding a column on mail.message should be done only if really necessary, and with as much minimal migration as possible.

Here we should avoid adding a column with default=True (millions of records to update). More in-depth I am really not in favor of adding a column on mail.message just for a corner case of a dummy bot people don't use anyway.

What is its purpose, isn't it possible to use another mechanism ?

Reply:
The idea was to have OdooBot send a chat message without opening the thread window, just showing the notification. I did not find another way to do this other than to implement the general feature.

I could use the kwargs from the `message_post` and look for an `open_thread` key in there, instead of adding the new column, or just scrap the feature altogether if it is not really worth it.

Reply:
Opening a thread window is a visual action that can be changed in the JS part only. Your solution is more flexible but is probably too overkill.

In the current code it would be such as overriding `_addNewMessagePostprocessThread` in `mail_bot`, however the current method is not really built to be overridden for this purpose (so it has to be refactored to allow overriding the `thread.detach` aspect while keeping the rest intact).